tutorials/integrations/nft-subsquid.md

```diff
@@ -9,250 +9,189 @@ _by Massimo Luraschi_
-    ```
+```js
+'https://v2.archive.subsquid.io/network/moonbase-substrate'
+```
 
 !!! note
-    The `lookupArchive` function is used to consult the [archive registry](https://github.com/subsquid/archive-registry){target=\_blank} and yield the archive address, given a network name. Network names should be in lowercase.
+    This code expects to find a working Moonbeam RPC URL in the `RPC_ENDPOINT` environment variable. You can get your own endpoint and API key from a supported [Endpoint Provider](/builders/get-started/endpoints/){target=\_blank}.
 
-You'll also need to modify the `Context` type so that it is exported and can be used in the `src/contract.ts` file.
+    Set it in the `.env` file and [Subsquid Cloud secrets](https://docs.subsquid.io/cloud/resources/env-variables/){target=\_blank} if and when you deploy your squid there. We tested the code using a public endpoint at `wss://wss.api.moonbeam.network`; we recommend using private endpoints for production.
 
-```ts
-export type Context = BatchContext<Store, Item>;
-```
+## Define the Batch Handler {: #define-batch-handler }
 
-Here is the end result:
+We'll need to rewrite the batch handler logic in the `src/main.ts` file.  We'll iterate over all of the events for each batch of blocks to find the EVM logs relative to the Exiled Racers contracts. We'll extract the from and to addresses and the token ID from the EVM logs. Then, we'll format this data as defined in the schema and save it to the database.
 
-```typescript
---8<-- 'code/tutorials/integrations/nft-subsquid/processor.ts'
-```
+Here is the result:
 
-!!! note
-    It is also worth pointing out that the `contract.tokenURI` call is accessing the **state** of the contract via a chain RPC endpoint. This is slowing down the indexing a little bit, but this data is only available this way. You'll find more information on accessing state in the [dedicated section of the Subsquid docs](https://docs.subsquid.io/substrate-indexing/evm-support#access-the-contract-state){target=\_blank}.
+```typescript title="src/main.ts"
+--8<-- 'code/tutorials/integrations/nft-subsquid/main.ts'
+```
 
 !!! note
-    This code expects to find a URL of a working Moonbeam RPC endpoint in the `RPC_ENDPOINT` environment variable. Set it in the `.env` file and in [Aquarium secrets](https://docs.subsquid.io/deploy-squid/env-variables){target=\_blank} if and when you deploy your squid there. We tested the code using a public endpoint available at `wss://wss.api.moonbeam.network`; for production, we recommend using [private endpoints](/builders/get-started/endpoints#endpoint-providers){target=\_blank}.
+    The `contract.tokenURI` call acccesses the **state** of the contract via a chain RPC endpoint. This can slow down indexing, but this data is only available in this way. You'll find more information on accessing state in the [dedicated section of the Subsquid docs](https://docs.subsquid.io/sdk/resources/tools/typegen/state-queries/#example-1){target=\_blank}.
```

Comment:
```suggestion
    The `contract.tokenURI` call accesses the **state** of the contract via a chain RPC endpoint. This can slow down indexing, but this data is only available in this way. You'll find more information on accessing state in the [dedicated section of the Subsquid docs](https://docs.subsquid.io/sdk/resources/tools/typegen/state-queries/#example-1){target=\_blank}.
```